File: problems/3.longest_substring-without-repeating-characters.c
int lengthOfLongestSubstring(char* s) {
    int set[255];
    for(int i = 0; i < 255; i++) {
        set[i] = 0;
    }

    int i, j, count;
    i = j = count = 0;
    while(s[j] != '\0') {
        if (!set[s[j]]) {
            set[s[j++]] = 1;
            if (j - i > count) {
                count = j - i;
            }
        } else {
            set[s[i++]] = 0;
        }
    }

    return count;
}

File: problems/other_langs/medium/3.longest_substring_without_repeating_characters.php
<?php
class Solution {

    /**
     * @param String $s
     * @return Integer
     */
    function lengthOfLongestSubstring($s) {
        $sarr = str_split($s);
        $maxCount = 0;
        $size = strlen($s);
        $i = $j = 0;

        while($i < $size && $j < $size) {
            if (!isset($temp[$s[$j]])) {
                $temp[$s[$j++]] = 0;
                if ($j - $i > $maxCount) {
                    $maxCount = $j - $i;
                }
            } else {
                unset($temp[$s[$i++]]);
            }
        }
        return $maxCount;
    }
}